cmd/grpcui/grpcui.go

```diff
@@ -358,11 +360,11 @@ func main() {
 		})
 	}
 
-	listener, err := net.Listen("tcp", fmt.Sprintf("127.0.0.1:%d", *port))
+	listener, err := net.Listen("tcp", fmt.Sprintf("%s:%d", *bind, *port))
 	if err != nil {
 		fail(err, "Failed to listen on port %d", *port)
 	}
-	fmt.Printf("gRPC Web UI available at http://127.0.0.1:%d/...\n", listener.Addr().(*net.TCPAddr).Port)
+	fmt.Printf("gRPC Web UI available at http://%s:%d/...\n", *bind, listener.Addr().(*net.TCPAddr).Port)
```

Comment:
If the bind address is `0.0.0.0` (bind to all interfaces), this should detect that and show `127.0.0.1` instead. Not sure if there is anything useful in `net` package to help detect this (other than string comparison) -- it would be nice if there were.

Reply:
yep, that's nice for usability. but two queries pls:

1. do you think it might be misleading? i.e. if user provides `-bind 0.0.0.0` and sees `available at http://127.0.0.1` wouldn't they be confused as to whether their option was picked up or not?
2. 0.0.0.0 is one case, but there are also [more possibilities](https://golang.org/pkg/net/#SplitHostPort) for binding such as `::` for IPv6, `::1`, `[::1]`, for localhost in IPv6, etc, and `net.Listen` supports all those bind options. but finding the right URL to print for the CLI user in all those cases is pretty tough, e.g. `http://[::1]:8080` would be as obscure as `http://0.0.0.0:8080`

rails/rack/puma/tomcat webservers print whatever the user specified directly ,such as `Listening on <whatever user entered>:<port>`, stepping out of the ambiguity, but I'm not sure if that can apply here...

thanks for the amazing project!!

Reply:
> do you think it might be misleading?

Not particularly. The intent is to provide a valid URL, so I'm more concerned about printing something that may not be useful. If you want a separate log line (maybe enabled with `-v`) that prints the bind address separately, that seems fine to eliminate any ambiguity here.

> ... but there are also more possibilities for binding ...

I'm not concerned about obscurity; my concern is just about printing a valid URL. So the address `http://[::1]:8080` is fine, since it would work fine in a browser.

Having said that, it appears that `http://0.0.0.0:8080` and `http://[::]:8080` *do actually work*, at least on my Macbook. What I'm not certain is whether that will work everywhere. My understanding was that, as a host address, 0.0.0.0 just means the default route. Will that always be localhost?

So... In summary, if 0.0.0.0 will always work as a valid URL and route to localhost, then I suppose I withdraw my objection.

Reply:
So far tested with curl and wget on the terminal, major mac browsers (chrome, FF, safari) and go `net/http`, java `URLConnection` and node `http.get` code on mac and linux (ubuntu 18.04). It was supported in all those places. unfortunately don't have a canonical link to show it tho, so will ping snippets here...

```
curl --head http://0.0.0.0:8000
HTTP/1.0 200 OK

wget http://0.0.0.0:8000
--2018-12-21 22:04:54--  http://0.0.0.0:8000/
Connecting to 0.0.0.0:8000... connected.
HTTP request sent, awaiting response... 200 OK

# java
new URL("http://0.0.0.0:8000").openConnection().getHeaderFields()

# node
const http = require('http');
http.get('http://0.0.0.0:8000', res => console.log(res));

# go
resp, err := http.Get("http://example.com/")
log.Println(resp)

# and major browsers...
```

I don't have a windows machine to test, but yes know for sure that chrome & firefox URL handling is same across platforms. So for most web development it works out of the box 👍

Reply:
Thanks for poking around! Sounds like my one concern is moot.

Reply:
Thanks!!